monitoring/alert/alert.go

```diff
@@ -0,0 +1,394 @@
+		if err := enablePolicies(os.Stdout, *projectID, true); err != nil {
+			log.Fatal(err)
+		}
+	case "disable":
+		if err := enablePolicies(os.Stdout, *projectID, false); err != nil {
+			log.Fatal(err)
+		}
+	default:
+		fmt.Fprintf(os.Stderr, "Invalid action: %q\n", flag.Arg(0))
+		flag.Usage()
+		os.Exit(1)
+	}
+}
+
+// [START monitoring_alert_list_policies]
+
+// listAlertPolicies lists the alert policies in the project.
+func listAlertPolicies(w io.Writer, projectID string) error {
+	ctx := context.Background()
+	client, err := monitoring.NewAlertPolicyClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	req := &monitoringpb.ListAlertPoliciesRequest{
+		Name: "projects/" + projectID,
+		// Filter:  "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+		// OrderBy: "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+	}
+	it := client.ListAlertPolicies(ctx, req)
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(w, "Name: %q\n", resp.GetName())
+		fmt.Fprintf(w, "Display Name: %q\n", resp.GetDisplayName())
+		fmt.Fprintf(w, "Documentation Content: %q\n\n", resp.GetDocumentation().GetContent())
+	}
+	return nil
+}
+
+// [END monitoring_alert_list_policies]
+
+// [START monitoring_alert_backup_policies]
+
+// backupPolicies writes a JSON representation of the project's alert
+// policies and notification channels.
+func backupPolicies(w io.Writer, projectID string) error {
+	b := backup{ProjectID: projectID}
+	ctx := context.Background()
+
+	alertClient, err := monitoring.NewAlertPolicyClient(ctx)
+	if err != nil {
+		return err
+	}
+	alertReq := &monitoringpb.ListAlertPoliciesRequest{
+		Name: "projects/" + projectID,
+		// Filter:  "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+		// OrderBy: "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+	}
+	alertIt := alertClient.ListAlertPolicies(ctx, alertReq)
+	for {
+		resp, err := alertIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		b.Alerts = append(b.Alerts, &alertPolicy{resp})
+	}
+
+	// [START monitoring_alert_list_channels]
+	channelClient, err := monitoring.NewNotificationChannelClient(ctx)
+	if err != nil {
+		return err
+	}
+	channelReq := &monitoringpb.ListNotificationChannelsRequest{
+		Name: "projects/" + projectID,
+		// Filter:  "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+		// OrderBy: "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+	}
+	channelIt := channelClient.ListNotificationChannels(ctx, channelReq)
+	for {
+		resp, err := channelIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		b.Channels = append(b.Channels, &channel{resp})
+	}
+	// [END monitoring_alert_list_channels]
+	e := json.NewEncoder(w)
+	if err := e.Encode(b); err != nil {
```

Comment:
```
if err := json.NewEncoder(w).Encode(b); err != nil {
```

But you might want to use json.MarshalIndent instead, especially since this is being printed out?

Reply:
Done.